src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventCounter.cs

```diff
@@ -95,6 +95,9 @@ internal void OnMetricWritten(double value)
             _count++;
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "CounterPayload type has been trimmer safe")]
```

Comment:
Instead of "CounterPayload type has been trimmer safe", it should be "The DynamicDependency will preserve the properties of CounterPayload"